packages/libp2p/src/circuit-relay/utils.ts

```diff
@@ -47,25 +48,33 @@ const doRelay = (src: Stream, dst: Stream, abortSignal: AbortSignal, limit: Requ
   const abortController = new AbortController()
   const signal = anySignal([abortSignal, abortController.signal])
 
-  const timeout = setTimeout(() => {
-    abortController.abort()
-  }, limit.duration)
+  let timeout: ReturnType<typeof setTimeout> | undefined
+
+  if (limit?.duration != null) {
+    timeout = setTimeout(() => {
+      abortController.abort()
+    }, limit.duration)
+  }
 
   let srcDstFinished = false
   let dstSrcFinished = false
 
-  const dataLimit = {
-    remaining: limit.data
+  let dataLimit: { remaining: bigint } | undefined
+
+  if (limit?.data != null) {
+    dataLimit = {
+      remaining: limit.data
```

Comment:
See comment above, If the function signature is changed then creating this object is unnecessary